src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java

```diff
@@ -104,6 +107,11 @@ public InteractionImpl(JDAImpl jda, DataObject data)
             }
             this.user = user;
         }
+
+        this.entitlements = data.optArray("entitlements").orElseGet(DataArray::empty)
+                .stream(DataArray::getObject)
+                .map(jda.getEntityBuilder()::createEntitlement)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .collect(Helpers.toUnmodifiableList());
```
Don't forget to add the import